nested-spire/README.md

```diff
@@ -0,0 +1,226 @@
+         - ./root/agent:/opt/spire/conf/agent
+         - /var/run/:/var/run/
+       command: ["-config", "/opt/spire/conf/agent/agent.conf"]
+```
+
+## Configuring NestedA SPIRE Deployment
+
+The same set of configurations are required for the `nestedB` SPIRE deployment but those changes are not described in the text to avoid needless repetition.
+
+SPIRE Agent and Server can be extended with a variety of [plugins](https://spiffe.io/spire/docs/extending/). The [UpstreamAuthority plugin](https://github.com/spiffe/spire/blob/master/doc/spire_server.md#built-in-plugins) type allows SPIRE Server to integrate with existing PKI systems. UpstreamAuthority plugins can sign certificates using CAs loaded from disk, third-party tools from AWS and Vault, and so on. Nested SPIRE deployments require the use of the [spire UpstreamAuthority plugin](https://github.com/spiffe/spire/blob/master/doc/plugin_server_upstreamauthority_spire.md) which uses an upstream SPIRE Server in the same trust domain to obtain intermediate signing certificates for SPIRE Server.
+
+The configuration file for the [nestedA-server](./nestedA/server/server.conf) includes the `spire` UpstreamAuthority plugin definition with the `root-server` as its upstream SPIRE Server.
+
+```console
+   UpstreamAuthority "spire" {
+ 	   plugin_data = {
+ 	       server_address      = "root-server"
+ 	       server_port         = 8081
+ 	       workload_api_socket = "/opt/spire/sockets/workload_api.sock"
+ 	   }
+    }
+```
+
+The Docker Compose definition for the `nestedA-server` service in the [docker-compose.yaml](docker-compose.yaml) file mounts the new local directory `sharedRootSocket` as a volume. Remember from the previous section that the `root-agent` socket is mounted on that directory. That way the `nestedA-server` can access the `root-agent` workload API and fetch its SVID.
+
+```console
+   nestedA-server:
+     # Share the host pid namespace so this server can be attested by the root agent
+     pid: "host"
+     image: gcr.io/spiffe-io/spire-server:0.10.1
+     hostname: nestedA-server
+     labels:
+       # label to attest nestedA-server against root-agent
+       - org.example.name=nestedA
+     volumes:
+       # Add root agent socket  
+       - ./shared/rootSocket:/opt/spire/sockets
+       - ./nestedA/server:/opt/spire/conf/server
+     command: ["-config", "/opt/spire/conf/server/server.conf"]
+```
+
+## Create Downstream Registration Entry
+
+The `nestedA-server` must be registered on the `root-server` to get SVIDs. We achieve this by creating a registration entry in the root SPIRE Server for the `nestedA-server`.
+
+```console
+   docker-compose exec -T root-server \
+       /opt/spire/bin/spire-server entry create \
+       -parentID "spiffe://example.org/spire/agent/x509pop/$(fingerprint root/agent/agent.crt.pem)" \
+       -spiffeID "spiffe://example.org/nestedA" \
+       -selector "docker:label:org.example.name:nestedA-server" \
+       -downstream \
+       -ttl 3600
```

Comment:
maybe for example purpose you can reduce that time? just to display people how downstream expires and is renewed?